plugins/inputs/opcua/opcua_client.go

```diff
@@ -476,7 +475,7 @@ func (o *OpcUA) setupOptions() error {
 		}
 	}
 
-	o.opts, err = generateClientOpts(endpoints, o.Certificate, o.PrivateKey, o.SecurityPolicy, o.SecurityMode, o.AuthMethod, o.Username, o.Password, time.Duration(o.RequestTimeout))
+	o.opts, err = o.generateClientOpts(endpoints, o.Certificate, o.PrivateKey, o.SecurityPolicy, o.SecurityMode, o.AuthMethod, o.Username, o.Password, time.Duration(o.RequestTimeout))
```

Comment:
If you are changing this anyway, you should maybe get rid of the struct-owned function parameters...

Reply:
Sure, done!